### PR TITLE
Make hypot available for all quantities instead of just SI lengths

### PR DESCRIPTION
**Pull request by adamreichold**

### CODE REVIEW

File: src/tests/quantity.rs [hunk 1]
```diff
@@ -452,5 +452,14 @@ mod float {
...
         }
+
+        #[cfg(feature = "std")]
+        quickcheck! {
+            #[allow(trivial_casts)]
+            fn hypot(l: V, r: V) -> bool {
+                Test::eq(&l.hypot(r),
+                    &f::Length::new::<meter>(l).hypot(f::Length::new::<meter>(r)).get::<meter>())
+            }
+        }
```
Comment (iliekturtles) [inline src/tests/quantity.rs:463]: Can you change these tests to use the normal (`f::`) base units and the kilometer/kilogram/kelvin (`k::`) base units. Something like the `ge` test  higher up in the file.

https://github.com/iliekturtles/uom/blob/afd4e515cbfb2c430f017af099f2cac9a05dd723/src/tests/quantity.rs#L172-L181

Comment (adamreichold) [reply src/tests/quantity.rs:463]: I think those tests do not really apply to the current signature
```
fn hypot(self, other: Self) -> Self;
```
explicitly prevents mixing base units. Do you want me to make this generic and switch to something like
```
fn hypot(&self, other: &Quantity<D, U1, V>) -> Self;
```
by inserting `change_base` into the implementation? Should this then also depend on the `autoconvert` feature?

Comment (iliekturtles) [reply src/tests/quantity.rs:463]: Good point, I missed that the signature currently requires the same base units. I think it should be switched to accept differing base units based on the `autoconvert` flag. Thoughts?

Comment (adamreichold) [reply src/tests/quantity.rs]: Add both variants of the signature depending on the `autoconvert` feature and extend the test to cover mixing base units.

**Comment by iliekturtles** [comment]
If you could apply some minor formatting changes from the diff below I'll merge!

```diff
diff --git a/src/quantity.rs b/src/quantity.rs
index 0aca8ae..b6aff6f 100644
--- a/src/quantity.rs
+++ b/src/quantity.rs
@@ -389,18 +389,18 @@ macro_rules! quantity {
             /// Calculates the length of the hypotenuse of a right-angle triangle given the legs.
             #[cfg(feature = "std")]
             #[inline(always)]
-            pub fn hypot<U1>(self, other: $quantity<U1, V>) -> Self
+            pub fn hypot<Ur>(self, other: $quantity<Ur, V>) -> Self
             where
                 V: $crate::num::Float,
-                U1: super::Units<V> + ?Sized,
+                Ur: super::Units<V> + ?Sized,
             {
                 Self {
                     dimension: $crate::lib::marker::PhantomData,
                     units: $crate::lib::marker::PhantomData,
-                    value: self.value.hypot(super::change_base::<Dimension, U, U1, V>(&other.value)),
+                    value: self.value.hypot(
+                        super::change_base::<Dimension, U, Ur, V>(&other.value)),
                 }
-            }
-            }
+            }}
 
             not_autoconvert! {
             /// Calculates the length of the hypotenuse of a right-angle triangle given the legs.
@@ -415,8 +415,7 @@ macro_rules! quantity {
                     units: $crate::lib::marker::PhantomData,
                     value: self.value.hypot(other.value),
                 }
-            }
-            }
+            }}
 
             /// Creates a struct that can be used to format a compatible quantity for display.
             ///
diff --git a/src/tests/quantities.rs b/src/tests/quantities.rs
index 6621cc9..2e5e62a 100644
--- a/src/tests/quantities.rs
+++ b/src/tests/quantities.rs
@@ -227,10 +227,8 @@ mod float {
         quickcheck! {
             #[allow(trivial_casts)]
             fn hypot(l: V, r: V) -> bool {
-                Test::eq(
-                    &l.hypot(r),
-                    &Length::new::<meter>(l).hypot(Length::new::<meter>(r)).get::<meter>(),
-                )
+                Test::eq(&l.hypot(r),
+                    &Length::new::<meter>(l).hypot(Length::new::<meter>(r)).get::<meter>())
             }
         }
     }
diff --git a/src/tests/quantity.rs b/src/tests/quantity.rs
index edf81da..d7ed7bd 100644
--- a/src/tests/quantity.rs
+++ b/src/tests/quantity.rs
@@ -457,10 +457,8 @@ mod float {
         quickcheck! {
             #[allow(trivial_casts)]
             fn hypot_same(l: V, r: V) -> bool {
-                Test::eq(
-                    &l.hypot(r),
-                    &f::Length::new::<meter>(l).hypot(f::Length::new::<meter>(r)).get::<meter>(),
-                )
+                Test::eq(&l.hypot(r),
+                    &f::Length::new::<meter>(l).hypot(f::Length::new::<meter>(r)).get::<meter>())
             }
         }
 
@@ -469,19 +467,13 @@ mod float {
         quickcheck! {
             #[allow(trivial_casts)]
             fn hypot_mixed(l: V, r: V) -> bool {
-                let fk = Test::approx_eq(
-                    &l.hypot(r),
-                    &f::Length::new::<meter>(l).hypot(k::Length::new::<meter>(r)).get::<meter>(),
-                );
-
-                let kf = Test::approx_eq(
-                    &l.hypot(r),
-                    &k::Length::new::<meter>(l).hypot(f::Length::new::<meter>(r)).get::<meter>(),
-                );
+                let fk = Test::approx_eq(&l.hypot(r),
+                    &f::Length::new::<meter>(l).hypot(k::Length::new::<meter>(r)).get::<meter>());
+                let kf = Test::approx_eq(&l.hypot(r),
+                    &k::Length::new::<meter>(l).hypot(f::Length::new::<meter>(r)).get::<meter>());
 
                 fk && kf
             }
-        }
-        }
+        }}
     }
 }
```

**Comment by adamreichold** [comment]
> If you could apply some minor formatting changes from the diff below I'll merge!

Done. ON a related note, I enabled the "Allow edits by maintainers" checkbox since personally, I have no issue with you doing things like renames or reformats as per the maintainer's prerogative.

**Comment by iliekturtles** [comment]
Thanks again for this PR!

I haven't used the "Allow edits by maintainers" before but will keep an eye on it for future PRs.

**Pull request merged**